clients/admin-ui/src/features/datamap/datamap.slice.ts

```diff
@@ -143,11 +181,10 @@ export const datamapSlice = createSlice({
     },
     // Load columns for the first time. If columns are already loaded, no-op.
     loadColumns(draftState, { payload }: PayloadAction<DatamapColumn[]>) {
-      if (draftState.columns) {
-        return;
-      }
-
-      draftState.columns = payload;
+      draftState.columns = mergeColumns(
+        draftState.columns as DatamapColumn[] | undefined,
```

Comment:
if you use the generics in my previous comment, you could remove the `as DatamapColumn[] | undefined`